core/templates/pages/splash-page/splash-page.component.ts

```diff
@@ -15,111 +15,135 @@
  */
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
 
-require('base-components/base-content.directive.ts');
+import splashConstants from 'assets/constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { WindowDimensionsService } from 'services/contextual/window-dimensions.service.ts';
+import { LoaderService } from 'services/loader.service.ts';
+import { UserService } from 'services/user.service';
 
-require('domain/utilities/url-interpolation.service.ts');
-require('services/contextual/window-dimensions.service.ts');
-require('services/site-analytics.service.ts');
-require('services/user.service.ts');
+export interface Testimonial {
+  quote: string,
+  studentDetails: string,
+  imageUrl: string,
+  imageUrlWebp: string,
+  borderPresent: boolean
+}
 
-// TODO(#9186): Change variable name to 'constants' once this file
-// is migrated to Angular.
-import splashConstants from 'assets/constants';
+@Component({
+  selector: 'splashPage',
+  templateUrl: './splash-page.component.html',
+  styleUrls: []
+})
+export class SplashPageComponent implements OnInit {
+  isWindowNarrow: boolean = false;
+  classroomUrlFragment: string;
+  classroomUrl :string;
+  displayedTestimonialId: number;
+  testimonialCount: number;
+  testimonials = [];
+  userIsLoggedIn: boolean = null;
+  constructor(
+    private siteAnalyticsService: SiteAnalyticsService,
+    private urlInterpolationService: UrlInterpolationService,
+    private windowDimensionService: WindowDimensionsService,
+    private windowRef: WindowRef,
+    private userService: UserService,
+    private loaderService: LoaderService,
+  ) {}
 
-angular.module('oppia').component('splashPage', {
-  template: require('./splash-page.component.html'),
-  controller: [
-    '$rootScope', '$translate', 'LoaderService', 'SiteAnalyticsService',
-    'UrlInterpolationService', 'UserService', 'WindowDimensionsService',
-    function(
-        $rootScope, $translate, LoaderService, SiteAnalyticsService,
-        UrlInterpolationService, UserService, WindowDimensionsService) {
-      var ctrl = this;
-      ctrl.getStaticImageUrl = function(imagePath) {
-        if (imagePath) {
-          return UrlInterpolationService.getStaticImageUrl(imagePath);
-        }
-      };
+  getStaticImageUrl(imagePath: string): string {
+    return this.urlInterpolationService.getStaticImageUrl(imagePath);
+  }
 
-      ctrl.onClickBrowseLessonsButton = function() {
-        SiteAnalyticsService.registerClickBrowseLessonsButtonEvent();
-        return false;
-      };
+  onClickBrowseLessonsButton():void {
+    this.siteAnalyticsService.registerClickBrowseLessonsButtonEvent();
+    this.windowRef.nativeWindow.location.href = this.classroomUrl;
+    return;
```

Comment:
No need for this return here. Ditto below.

Reply:
done.